src/Ryujinx.Tests/Graphics/AstcDecoderTests.cs

```diff
@@ -0,0 +1,198 @@
+
+            // The decode function said it was valid data and that it could parse it.
+            Assert.AreEqual(true, succeeded);
+            // Length is the same as the one we made w/ ARM's decoder. That's good.
+            Assert.AreEqual(decodedRef.Length, outputBuffer.Length);
+
+            var wordsRef = RgbaWord.FromBytes(decodedRef.ToArray());
+            var wordsOut = RgbaWord.FromBytes(outputBuffer);
+            var wordDifferences = wordsRef.Select((x, i) => new { index = i, diff = x.Diff(wordsOut[i]) }).ToArray();
+
+            // BUT compression is funny.
+            // Calculate the byte differences. 
+            var byteDifferences = decodedRef.ToArray().Select((x, i) => new { index = i, delta = x - outputBuffer[i] }).ToList();
+
+            var matchCount = byteDifferences.Count(x => x.delta == 0);
+            var matchPercent = ((float)matchCount / outputBuffer.Length);
+
+            var wordUnchangedCount = wordDifferences.Count(x => x.diff.IsZero());
+            var wordUnchangedPercent = (float)wordUnchangedCount / wordDifferences.Length;
+
+            TestContext.Out.WriteLine($"Pixel-wise comparison: {wordUnchangedPercent * 100:F4} ({wordUnchangedCount}/{wordDifferences.Length})");
+            TestContext.Out.WriteLine($"Byte-wise comparison: {matchPercent * 100:F4} ({matchCount}/{byteDifferences.Count}) were same.");
+
+            for (var threshold = 1; threshold < 32; threshold++)
+            {
+                var tc = byteDifferences.Count(x => Math.Abs(x.delta) >= threshold);
+                var tcp = ((float)tc / byteDifferences.Count);
+                if (tc > 0)
+                    TestContext.Out.WriteLine($"{tcp * 100:F4}% ({tc}/{byteDifferences.Count}) are different by at least {threshold}.");
+            }
+
+            Assert.IsTrue(byteDifferences.All(x => Math.Abs(x.delta) <= 1));
+        }
+
+        /// <summary>
+        /// Get test data from FS using short name naming convention.
+        /// </summary>
+        /// <param name="shortName"></param>
+        /// <returns></returns>
+        private (ReadOnlyMemory<byte>, ReadOnlyMemory<byte>) GetTestDataTupleFromShortname(string shortName, int blockWidth, int blockHeight)
+        {
+            var encodedRef = GetFileDataFromPath($"{shortName}.l-{blockWidth}x{blockHeight}-100.astc");
+            // var decodedRef = _getFileDataFromPath($"{shortName}.s4x4.astc.png");
+            var rgba8raw = GetFileDataFromPath($"{shortName}.l-{blockWidth}x{blockHeight}-100.astc.rgba");
+
+            return (encodedRef, rgba8raw);
+        }
+
+        private ReadOnlyMemory<byte> GetFileDataFromPath(string relativeFilePath)
+        {
+            var fullPath = Path.Join(_testDataDir, relativeFilePath);
+            return File.ReadAllBytes(fullPath);
+        }
+
+        private class RgbaWord
+        {
+            public byte r;
```

Comment:
```suggestion
            public byte R;
```
Same for the other ones.